src/python/grpcio/grpc/_simple_stubs.py

```diff
@@ -178,33 +192,51 @@ def get_channel(
             channel_data = self._mapping.get(key, None)
             if channel_data is not None:
                 channel = channel_data[0]
+                registered_method_handles = channel_data[2]
+                # Register a new call handle if we're calling a registered method for an
+                # existing channel and this method is not registered.
+                if (
+                    _registered_method
+                    and method not in registered_method_handles.keys()
```

Comment:
It seems odd that we're doing this book-keeping both at the `ChannelCache` level and at the `Channel` level. Why can't this block of added code just become

```
call_handle = channel._get_registered_call_handle(method)
```